### PR TITLE
Dev/mapping test server

### PR DESCRIPTION
**Pull request by friso**
This adds a minimal server that exposes a endpoint for testing mappings and schema. While this addition is not very large, it required quite a few changes in the way the mapping and processing organise their data. As a result of this change, the mapping doesn't use the query string from incoming requests anymore, but instead uses certain HttpServerExchange attachments, which are set by the client side cookie endpoint.

### CODE REVIEW

File: src/scripts/mapping-test-server [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+# Locate config dir and source divolte-env.sh
+[ -n "$DIVOLTE_CONF_DIR" ] && CONF_DIR="$DIVOLTE_CONF_DIR" || CONF_DIR="$APP_HOME/conf"
+
+# add libs to CLASSPATH
+function join { local IFS="$1"; shift; echo "$*"; }
+CLASSPATH=$(join ":" $APP_HOME/lib/*.jar)
+
+# attempt to find java
+if [ -z "$JAVA_HOME" ]; then
+  for candidate in \
```
Comment (asnare) [inline src/scripts/mapping-test-server]: Detecting Java is annoyingly difficult. There are some corner cases here, such as:
 - Do we care which Java 8 version to use, if there are multiple? Right now we'll choose the first, according to lexical sorting, whereas we probably want the latest.
 - Should we really choose Java 8 in preference to whatever `/usr/java/default` is?

We also never verify if the user's `JAVA_HOME` value is correct.

Given this, I'm tempted to propose an alternate strategy:
 - If `JAVA_HOME` is set and valid, use it.
 - If `JAVA_HOME` is not set, java has to be on the path.

File: src/main/java/io/divolte/server/MappingTestServer.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        final HttpHandler handler = new AllowedMethodsHandler(this::handleEvent, Methods.POST);
+        undertow = Undertow.builder()
+                .addHttpListener(port, host)
+                .setHandler(handler)
+                .build();
+    }
+
+    private Schema loadSchema(final String schemaFilename) throws IOException {
+        final Parser parser = new Schema.Parser();
+        final String schemaFileName = schemaFilename;
```
Comment (asnare) [inline src/main/java/io/divolte/server/MappingTestServer.java]: Um?
